src/Overlay/Modal/Modal.tsx

```diff
@@ -7,10 +7,19 @@ type ModalProps = {
     header?: React.ReactNode;
     children?: React.ReactNode;
     footer?: React.ReactNode;
+    closeProps?: CloseProps;
+    hideCancelButton: boolean;
   };
 
+type CloseProps = {
```

Comment:
closeProps will also need to forward all Button props and types. So for example we could set `style` or `onClick` for the button or any other support prop. You won't be able to write out every single one so it will be extend of ButtonProps. And theses props will need to be spread across the button

Reply:
Sure, fixed.